orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy

```diff
@@ -84,7 +84,10 @@ trait DeploymentDetailsAware {
   List<Stage> getAncestors(Stage stage, Execution execution) {
     if (stage?.requisiteStageRefIds) {
       def previousStages = execution.stages.findAll {
-        it.refId in stage.requisiteStageRefIds
+        // Include cloudProvider check to avoid confusion with multi-provider,
+        // in some cases ancestors can be one of any multi-provider
+        // Eg parent->aws and child->titus
+        it.refId in stage.requisiteStageRefIds && it.context.cloudProvider == stage.context.cloudProvider
```

Comment:
this makes sense. But.. I am still confused why this is necessary since the assumption is that we find the nearest ancestor, right?

With this fix, what happens in this case:

1. Parent (aws)
    a. Find Image
    b. Run Child Pipeline
2. Child (titus)
    a. Find Image
    b. Deploy
3. Grandchild (titus)
    a. Find Image
    b. Deploy

will 3b deploy image from 3a or 2a?

Reply:
For the above case it deploys the image from 3a , i tested a similar setup with 3rd pipeline being an aws find image and deploy which works fine. I'm going to test a few more cases before merging this change.